README.rst

```diff
@@ -3,56 +3,71 @@ openedx-completion-aggregator
 
 |pypi-badge| |travis-badge| |codecov-badge| |pyversions-badge| |license-badge|
 
-openedx-completion-aggregator is a Django app that aggregates block level
-completion data for different block types for Open edX.
+openedx-completion-aggregator is a Django app that aggregates block level completion data for different block types for Open edX.
 
-Overview
---------
+What does that mean?
 
-openedx-completion-aggregator uses the pluggable django app pattern to
-ease installation.  To use in edx-platform, do the following:
+A standard Open edX installation can track the completion of individual XBlocks in a course, which is done using the `completion library <https://github.com/edx/completion#completion>`_. This completion tracking is what powers the green checkmarks shown in the course outline and course navigation as the learner completes each unit in the course:
 
-1.  Install the app into your virtualenv.
+.. image:: docs/completion.png
+   :width: 100%
 
-    ..code_block::
+When completion tracking is enabled (and green checkmarks are showing, as seen above), it is only tracked at the XBlock level. You can use the Course Blocks API to check the completion status of any individual XBlock in the course, for a single user. For example, to get the completion of the XBlock with usage ID ``block-v1:OpenCraft+completion+demo+type@html+block@demo_block`` on the LMS instance ``courses.opencraft.com`` by the user ``MyUsername``, you could call this REST API::
+
+    GET https://courses.opencraft.com/api/courses/v1/blocks/block-v1:OpenCraft+completion+demo+type@html+block@demo_block?username=MyUsername&requested_fields=completion
+
+The response will include a ``completion`` value between ``0`` and ``1``.
+
+However, what if you want to know the overall % completion of an entire course? ("Alex, you have completed 45% of Introduction to Statistics") Or what if you as an instructor want to get a report of how much of Section 1 every student in a course has completed? Those queries are either not possible or too slow using the APIs built in to the LMS and ``completion``.
+
+This Open edX plugin, ``openedx-completion-aggregator`` watches course activity and asynchronously updates database tables with "aggregate" completion data. "Aggregate" data means completion data summed up over all XBlocks into a course and aggregated at higher levels, like the subsection, section, and course level. The completion aggregator provides a REST API that can provide near-instant answers to queries such as:
+
+* What % complete are each of the courses that I'm enrolled in?
+* What % of each section in Course X have my students completed?
+* What is the average completion % among all enrolled students in a course?
```

Comment:
Should we should clarify that aggregate queries like these will only be as accurate as the latest data based ont he last aggregation run?

Reply:
Sure, added a sentence.

> (Note that answers to these "aggregate" questions may be slightly out of date, because they are computed asynchronously. How often they are updated is configurable.)